Segmentation fault of client tool when input LDIF line is splitted but not indented correctly.

Resolves: #716855
Author: Howard Chu <hyc@openldap.org>
Upstream: ITS#6978, git 4b057bc

diff --git a/clients/tools/ldapmodify.c b/clients/tools/ldapmodify.c
index 7cd9774..2d11045 100644
--- a/clients/tools/ldapmodify.c
+++ b/clients/tools/ldapmodify.c
@@ -449,7 +449,7 @@ process_ldif_rec( char *rbuf, int linenum )
 			fprintf( stderr, _("%s: invalid format (line %d) entry: \"%s\"\n"),
 				prog, linenum+i, dn == NULL ? "" : dn );
 			rc = LDAP_PARAM_ERROR;
-			break;
+			goto leave;
 		}
 		freeval[i] = freev;
 
